code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -190,27 +236,108 @@ void setName(Map<String, Object> inspect, Map<String, Object> data, Instance ins
     }
 
     void setNetwork(Map<String, Object> inspect, Map<String, Object> data, Instance instance) {
-        Network network = null;
-        Network rancherNetwork = networkDao.getNetworkForObject(instance, KIND_HOSTONLY);
-        String ip = getDockerIp(inspect);
-        if (StringUtils.isNotEmpty(ip) && isIpInNetwork(ip, rancherNetwork)) {
-            DataAccessor.fields(instance).withKey(FIELD_REQUESTED_IP_ADDRESS).set(ip);
-            network = rancherNetwork;
+        String networkMode = checkNoneNetwork(inspect);
+        if (setNetworkModeIfNotNull(instance, networkMode))
```

Comment:
I can see that the setNetworkModeIfNotNull returns true only when networkMode is not null. Do you think it would be more clear if we call this method only once, and only after networkMode is resolved to not null in the caller stack?